### PR TITLE
fix(ui): backend support for ui update config key

### PR DESCRIPTION
**Pull request by SuperRxan**
Signed-off-by: Jianxiang Ran <rxan_embedded@163.com>

### CODE REVIEW

File: internal/server/rest.go [hunk 1]
```diff
@@ -131,13 +129,11 @@ func createRestServer(ip string, port int, needToken bool) *http.Server {
 	r.HandleFunc("/metadata/sources/{name}", sourceMetaHandler).Methods(http.MethodGet)
 	r.HandleFunc("/metadata/sources/yaml/{name}", sourceConfHandler).Methods(http.MethodGet)
 	r.HandleFunc("/metadata/sources/{name}/confKeys/{confKey}", sourceConfKeyHandler).Methods(http.MethodDelete, http.MethodPost)
-	r.HandleFunc("/metadata/sources/{name}/confKeys/{confKey}/field", sourceConfKeyFieldsHandler).Methods(http.MethodDelete, http.MethodPost)
```
Comment (ngjaying) [inline internal/server/rest.go:134]: POST or PUT?

File: internal/server/rest.go [hunk 2]
```diff
@@ -131,13 +129,11 @@ func createRestServer(ip string, port int, needToken bool) *http.Server {
...
 
 	r.HandleFunc("/metadata/connections", connectionsMetaHandler).Methods(http.MethodGet)
 	r.HandleFunc("/metadata/connections/{name}", connectionMetaHandler).Methods(http.MethodGet)
 	r.HandleFunc("/metadata/connections/yaml/{name}", connectionConfHandler).Methods(http.MethodGet)
 	r.HandleFunc("/metadata/connections/{name}/confKeys/{confKey}", connectionConfKeyHandler).Methods(http.MethodDelete, http.MethodPost)
-	r.HandleFunc("/metadata/connections/{name}/confKeys/{confKey}/field", connectionConfKeyFieldsHandler).Methods(http.MethodDelete, http.MethodPost)
```
Comment (ngjaying) [inline internal/server/rest.go:140]: POST or PUT?

File: internal/topo/topo.go [hunk 3]
```diff
@@ -154,7 +154,7 @@ func (s *Topo) Open() <-chan error {
 		defer s.mu.Unlock()
 		var err error
 		if s.store, err = state.CreateStore(s.name, s.qos); err != nil {
-			fmt.Println(err)
```
Comment (ngjaying) [inline internal/topo/topo.go:157]: Why changing this?

**Review by ngjaying — COMMENTED** [review]
The UI will use delete->post to update a confkey?

File: internal/server/rest.go (hunk 1 above)
Comment (SuperRxan) [reply internal/server/rest.go:134]: using post to create new configkey and update configkey

File: internal/topo/topo.go (hunk 3 above)
Comment (SuperRxan) [reply internal/topo/topo.go:157]: this can not print the error info context , do not know which rule have what error

**Review by ngjaying — APPROVED** [review]

**Pull request merged**